org.opentosca.planbuilder/src/org/opentosca/planbuilder/AbstractBuildPlanBuilder.java

```diff
@@ -22,86 +22,95 @@
 import org.opentosca.planbuilder.model.tosca.AbstractTopologyTemplate;
 import org.opentosca.planbuilder.utils.Utils;
 
-
 public abstract class AbstractBuildPlanBuilder extends AbstractPlanBuilder {
 
-	public AbstractPlan generatePOG(final String id, final AbstractDefinitions definitions, final AbstractServiceTemplate serviceTemplate, Collection<AbstractNodeTemplate> nodeTemplates, Collection<AbstractRelationshipTemplate> relationshipTemplates) {
+	public AbstractPlan generatePOG(final String id, final AbstractDefinitions definitions,
+			final AbstractServiceTemplate serviceTemplate, Collection<AbstractNodeTemplate> nodeTemplates,
+			Collection<AbstractRelationshipTemplate> relationshipTemplates) {
 		Collection<AbstractActivity> activities = new ArrayList<AbstractActivity>();
 		Set<Link> links = new HashSet<Link>();
-		
+
 		Map<AbstractNodeTemplate, AbstractActivity> nodeMapping = new HashMap<AbstractNodeTemplate, AbstractActivity>();
 		Map<AbstractRelationshipTemplate, AbstractActivity> relationMapping = new HashMap<AbstractRelationshipTemplate, AbstractActivity>();
-				
-		this.generatePOGActivitesAndLinks(activities, links, nodeMapping, nodeTemplates, relationMapping, relationshipTemplates);
-				
-		//this.cleanLooseEdges(links);
-		
-		AbstractPlan plan = new AbstractPlan(id, AbstractPlan.PlanType.BUILD, definitions, serviceTemplate, activities, links) {
-			
+
+		this.generatePOGActivitesAndLinks(activities, links, nodeMapping, nodeTemplates, relationMapping,
+				relationshipTemplates);
+
+		// this.cleanLooseEdges(links);
+
+		AbstractPlan plan = new AbstractPlan(id, AbstractPlan.PlanType.BUILD, definitions, serviceTemplate, activities,
+				links) {
+
 		};
 		return plan;
 	}
-	
+
 	private void cleanLooseEdges(Map<AbstractActivity, AbstractActivity> links) {
 		List<AbstractActivity> keysToRemove = new ArrayList<>();
-		
-		for(AbstractActivity key : links.keySet()) {
-			if(key == null) {
+
+		for (AbstractActivity key : links.keySet()) {
+			if (key == null) {
 				keysToRemove.add(key);
-			} else if(links.get(key) == null) {
+			} else if (links.get(key) == null) {
 				keysToRemove.add(key);
 			}
 		}
-		
+
 		keysToRemove.forEach(key -> links.remove(key));
```

Comment:
could be changed into a full-blown stream operation by streaming the keySet and filtering onto the keys like so:

```
links.keySet().stream()
    .filter(candidate -> candidate == null || links.get(candidate) == null)
    .forEach(links::remove);
```

Which in turn can be made more expressive by changing it to:

    links.remove(null);
    for (Iterator<AbstractActivity> vals = links.values().iterator(); vals.hasNext(); ) {
       if (vals.next() == null) {
           vals.remove();
        }
    }